### DISCUSSION TITLE
oidc_get_absolute_url using redirect_uri?

### DISCUSSION THREAD

**Discussion by brandonk10** [OpenIDC/mod_auth_openidc#1092] [Q&A] [answered]
I've been working against the latest master and have question about the new logic to allow relative URLs.
```
const char *oidc_get_absolute_url(request_rec *r, oidc_cfg *cfg, const char *url) {
	if ((url != NULL)
			&& (url[0] == OIDC_CHAR_FORWARD_SLASH)) {
		url = apr_pstrcat(r->pool, oidc_get_current_url_base(r, cfg->x_forwarded_headers),
				cfg->redirect_uri, NULL);

		oidc_debug(r, "determined absolute url: %s", url);
	}
	return url;
}
```
It looks like this method is always appending the redirect_uri to the base URL, but I think this is getting used on default URI and logged out URI as well.  Should the reference to redirect_uri be using the url param for the end of the concatenation?

**Comment by zandbelt** [comment, marked as answer]
you are correct, that work was not finished quite yet and this is a bug indeed; thanks for pointing it out, see https://github.com/OpenIDC/mod_auth_openidc/commit/3fe626e9df54b077e6e9274a986b29040da3f09f
